django_loci/tests/base/test_selenium.py

```diff
@@ -0,0 +1,74 @@
+from django.urls.base import reverse
+from selenium.webdriver import ActionChains
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import Select, WebDriverWait
+
+from openwisp_utils.tests import SeleniumTestMixin
+
+from .. import TestAdminMixin, TestLociMixin
+
+
+class BaseTestAdminSelenium(SeleniumTestMixin, TestAdminMixin, TestLociMixin):
+    def _get_url_prefix(self):
+        return '{0}_{1}'.format(
+            self.object_url_prefix, self.object_model.__name__.lower()
+        )
+
+    @classmethod
+    def _get_prefix(cls):
+        s = '{0}-{1}-content_type-object_id'
+        return s.format(
+            cls.location_model._meta.app_label,
+            cls.object_location_model.__name__.lower(),
+        )
+
+    @property
+    def add_url(self):
+        return '{0}_add'.format(self._get_url_prefix())
```

Comment:
This seems duplicated from `BaseTestAdminInline`. Can we create a mixin to centralize the logic?

Reply:
Yes that would work.